python/rikai/spark/sql/codegen/mlflow_logger.py

```diff
@@ -44,6 +48,37 @@ def __init__(self, flavor: str):
         self.spec_version = MlflowLogger._CURRENT_MODEL_SPEC_VERSION
         self.flavor = flavor
 
+    def _log_tensorflow_model(
+        self,
+        model: Any,
+        artifact_path: str,
+        registered_model_name: str,
+        **kwargs,
+    ):
+        import tensorflow as tf
+
+        try:
+            import mlflow
+        except ImportError as e:
+            raise ImportError(
+                "Couldn't import mlflow. Please make sure to "
+                "`pip install mlflow` explicitly or install "
+                "the correct extras like `pip install rikai[mlflow]`"
+            ) from e
+
+        with tempfile.TemporaryDirectory() as tmp_dir:
+
+            tf.saved_model.save(model, tmp_dir)
```

Comment:
you had to do this because the default mlflow behavior doesn't return a runnable model?

Reply:
Yea, the parameters of `mlflow.tensorflow.<ANYTHING>` is super weird .

For example, the first parameter of https://mlflow.org/docs/latest/python_api/mlflow.tensorflow.html#mlflow.tensorflow.log_model is a path.  I dont really know what do users expect from mlflow tbh.

Reply:
And `mlflow.tensorflow.load_model` does not actually return model, but returns model signatures 🤷